### PR TITLE
feat: check code up to date when build or release

### PR DESCRIPTION
**Pull request by michael8512**
## What type of PR is this?

- [x] Feature
- [ ] Bugfix
- [ ] Test
- [ ] Documentation
- [ ] Refactor
- [ ] Style
- [ ] Chore

## Which issue(s) this PR fixes:

Fixes #

## Does this PR introduce a user interface change?

- [ ] Yes(screenshot is required)
- [x] No

### CODE REVIEW

File: cli/lib/build.js [hunk 1]
```diff
@@ -52,6 +52,54 @@ const checkBranch = async () => {
...
+      type: 'confirm',
+      name: 'updateCode',
+      message: 'Whether to auto update the code? Or you can update manually and then select "N" to continue.',
+      default: false,
+    },
+  ]);
+
+  if (answer.updateCode) {
+    logInfo(`performing "git pull upstream develop" inside "${rootDir}"`);
+    await execSync(`git pull upstream develop`, { cwd: rootDir });
```
Comment (McDaddy) [inline cli/lib/build.js:69]: not every has the default upstream name as `upstream`, e.g. I rename it as `erda-source`, then if I run this script, I would get error

Comment (McDaddy) [reply cli/lib/build.js:69]: image

File: cli/lib/build.js [hunk 2]
```diff
@@ -52,6 +52,54 @@ const checkBranch = async () => {
...
+    }
+    logInfo(`performing "git pull upstream master" inside "${enterpriseDir}"`);
+    await execSync(`git pull upstream master`, { cwd: enterpriseDir });
+
+    answer = await inquirer.prompt([
+      {
+        type: 'confirm',
+        name: 'solvedConflicts',
+        message: 'Have all merge conflicts been solved?',
+        default: false,
```
Comment (daskyrk) [inline cli/lib/build.js]: You can use a package such as `git-status` to check if there is any conflict or uncommitted changes.

File: cli/lib/build.js [hunk 3]
```diff
@@ -143,27 +191,44 @@ const checkModuleValid = async (execPath)=> {
   }
 }
 
+const checkRebuildModules = async () => {
+  if (fs.existsSync(path.resolve(publicDir, 'static'))) {
```
Comment (McDaddy) [inline cli/lib/build.js:195]: Reuse local public resource for new build.... I don't think it's a good idea.
e.g. member A fully built for commit A(change shell code) then publish to version, later member B fully built for commit B(change core code) and publish to version, finally member A partially built for commit C(change shell code), but depends on the built of commit A. Which would drop commit B feature.
Another problem is if I don't release project for more than three days, then I don't know whether my local resource is still valid, so I have to make full building again.
Anyway, this pr can merge for now. I'll make adjustment later.

**Review by McDaddy — APPROVED** [review]

**Pull request merged**